### PR TITLE
support embed command in dotenv file

### PR DESCRIPTION
**Pull request by DarthPestilane**
Support embed commands via `$()` (not supported on Windows)

```env
START_TIME=$(date)
```

Note that using $() might not work depending on your shell.

### CODE REVIEW

**Comment by hairyhenderson** [comment]
IMO it would be best if this were an opt-in feature, since this can open up all kinds of security risks!

**Comment by DarthPestilane** [comment]
Agreed.

**Comment by DarthPestilane** [comment]
So how about I do it this way:

```go
godotenv.EnableEmbed()
godotenv.Load()
```

If we don't `EnableEmbed()` or `DisableEmbed()` before `Load()`, godotenv's not gonna parse the embed.

File: godotenv.go [hunk 1]
```diff
@@ -308,6 +318,16 @@ func parseValue(value string, envMap map[string]string) string {
 		}
 	}
 
+	// embed commands
+	if shouldParseEmbed {
+		if strings.HasPrefix(value, "$(") && strings.HasSuffix(value, ")") {
+			cmdStr := value[2 : len(value)-1]
+			cmd := exec.Command("/bin/sh", "-c", cmdStr)
+			out, _ := cmd.Output()
```
Comment (durandj) [inline godotenv.go]: Why silence the error here? Wouldn't that make it harder to diagnose failures?

File: godotenv.go [hunk 2]
```diff
@@ -308,6 +318,16 @@ func parseValue(value string, envMap map[string]string) string {
 		}
 	}
 
+	// embed commands
+	if shouldParseEmbed {
+		if strings.HasPrefix(value, "$(") && strings.HasSuffix(value, ")") {
+			cmdStr := value[2 : len(value)-1]
+			cmd := exec.Command("/bin/sh", "-c", cmdStr)
```
Comment (durandj) [inline godotenv.go]: Shouldn't you do a check for the OS here and if on Windows create a PowerShell session?

**Comment by the-wondersmith** [comment]
@durandj @DarthPestilane seeing as it looks like this went stale, I've addressed the comments / concerns and opened a new PR #230

**Comment by joho** [comment]
Closing per https://github.com/joho/godotenv/issues/182 but thank you

**Pull request closed without merging**